### ISSUE TITLE
Improve osquery integration

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#1348]
We propose these changes to the osquery integration:

- [x] Alert will never include null or empty values. 1ab55ddc1
- [x] Rule description should include the name of the query (if present). ad3369e5b
- [x] Create specific rules with a group for predefined osquery's packs. a0a8cc1a8
- [x] Extract the name of the pack as a new parameter. b3aaf3a26
- [x] The integration will print an alert if `<run_daemon>` is disabled since the agent won't be able to reload the configuration or apply decorators and extra packs. (Already done as `INFO`).
- [x] Osquery events should have a "root" object like `osquery.` in order to prevent data type collision in the App. 53b9b82c2

Will work on branch _3.7-osquery-improvements_.

**Comment by vikman90** [comment]
This issue depends on #1350.

**Comment by vikman90** [comment]
## osquery-monitor pack example

```
{
  "timestamp": "2018-09-19T08:35:40.530+0200",
  "rule": {
    "level": 4,
    "description": "osquery event: pack_osquery-monitoring_osquery_info",
    "id": "24020",
    "mail": false,
    "groups": [
      "osquery",
      "osquery_monitoring"
    ]
  },
  "agent": {
    "id": "000",
    "name": "stretch"
  },
  "manager": {
    "name": "stretch"
  },
  "id": "1537338940.2833",
  "full_log": "{\"name\":\"pack_osquery-monitoring_osquery_info\",\"hostIdentifier\":\"stretch\",\"calendarTime\":\"Wed Sep 19 06:43:13 2018 UTC\",\"unixTime\":1537339393,\"epoch\":0,\"counter\":0,\"decorations\":{\"host_uuid\":\"6AF93292-17E0-44F5-B321-35EA577BCC3F\",\"username\":\"root\"},\"columns\":{\"build_distro\":\"xenial\",\"build_platform\":\"ubuntu\",\"config_hash\":\"978d0b177bfa74840c676f0ea7898309c525b6f8\",\"config_valid\":\"1\",\"counter\":\"43\",\"extensions\":\"active\",\"instance_id\":\"71bf7c39-6a89-4228-bc12-359115b79a21\",\"pid\":\"11871\",\"resident_size\":\"17976000\",\"start_time\":\"1537339376\",\"system_time\":\"20\",\"user_time\":\"60\",\"uuid\":\"6AF93292-17E0-44F5-B321-35EA577BCC3F\",\"version\":\"3.3.0\",\"watcher\":\"11868\"},\"action\":\"added\"}",
  "decoder": {
    "name": "json"
  },
  "data": {
    "action": "added",
    "name": "pack_osquery-monitoring_osquery_info",
    "hostIdentifier": "stretch",
    "calendarTime": "Wed Sep 19 06:43:13 2018 UTC",
    "unixTime": "1537339393",
    "epoch": "0",
    "counter": "0",
    "decorations": {
      "host_uuid": "6AF93292-17E0-44F5-B321-35EA577BCC3F",
      "username": "root"
    },
    "columns": {
      "build_distro": "xenial",
      "build_platform": "ubuntu",
      "config_hash": "978d0b177bfa74840c676f0ea7898309c525b6f8",
      "config_valid": "1",
      "counter": "43",
      "extensions": "active",
      "instance_id": "71bf7c39-6a89-4228-bc12-359115b79a21",
      "pid": "11871",
      "resident_size": "17976000",
      "start_time": "1537339376",
      "system_time": "20",
      "user_time": "60",
      "uuid": "6AF93292-17E0-44F5-B321-35EA577BCC3F",
      "version": "3.3.0",
      "watcher": "11868"
    }
  },
  "location": "osquery"
}
```

## Global query example

This alerts comes from a custom query named "deb_packages":

```
{
  "timestamp": "2018-09-19T08:35:40.530+0200",
  "rule": {
    "level": 3,
    "description": "osquery event: deb_packages",
    "id": "24010",
    "mail": false,
    "groups": [
      "osquery"
    ]
  },
  "agent": {
    "id": "000",
    "name": "stretch"
  },
  "manager": {
    "name": "stretch"
  },
  "id": "1537338940.582079",
  "full_log": "{\"name\":\"deb_packages\",\"hostIdentifier\":\"stretch\",\"calendarTime\":\"Wed Sep 19 06:50:15 2018 UTC\",\"unixTime\":1537339815,\"epoch\":0,\"counter\":0,\"decorations\":{\"host_uuid\":\"6AF93292-17E0-44F5-B321-35EA577BCC3F\",\"username\":\"root\"},\"columns\":{\"arch\":\"amd64\",\"name\":\"iproute2\",\"revision\":\"1+deb9u1\",\"size\":\"1757\",\"source\":\"\",\"version\":\"4.9.0-1+deb9u1\"},\"action\":\"added\"}",
  "decoder": {
    "name": "json"
  },
  "data": {
    "action": "added",
    "name": "deb_packages",
    "hostIdentifier": "stretch",
    "calendarTime": "Wed Sep 19 06:50:15 2018 UTC",
    "unixTime": "1537339815",
    "epoch": "0",
    "counter": "0",
    "decorations": {
      "host_uuid": "6AF93292-17E0-44F5-B321-35EA577BCC3F",
      "username": "root"
    },
    "columns": {
      "arch": "amd64",
      "name": "iproute2",
      "revision": "1+deb9u1",
      "size": "1757",
      "version": "4.9.0-1+deb9u1"
    }
  },
  "location": "osquery"
}
```
